docs/GUIDE.md

```diff
@@ -0,0 +1,137 @@
+```xml
+<key>NSCameraUsageDescription</key>
+<string>camera for self capture</string>
+<key>NSMicrophoneUsageDescription</key>
+<string>mic for my voice</string>
+```
+  
+![](/docs/images/blog_avp5_plist.png "plist")
+
+You should confirm your permission to allow the app to use your VisionPro’s camera and microphone. Once you join the channel, you should see yourself in your avatar form and the remote user should show up on your screen!
+
+![](/docs/images/blog_avp6_device.gif "device")
+
+The Web user can also see you in your digital form and you can start chatting together now!
+
+![](/docs/images/blog_avp7_webuser.gif "web user")
+  
+
+## Basic Code Walkthrough
+
+Reading and understanding the code is never a boring task for developers! Here is a break-down of the project source so you can find the area of your interest quickly.
+
+|Class|Purpose|Notes|
+| --------   | -----:  | :----:  |
+Agora_VisionApp|Entry point of application|Standard entry point
+ContentView|UI View presenting to user at the beginning|Accepts the meeting’s channel name. Calls VisionVideoCallView
+VisionVideoCallView|Showing user’s video feeds as a grid format|Input the appID here
+AgoraVideoCanvasView|The video canvas view for rendering the video|This is a reusable class that works on iOS/MacOS as well.
+AgoraManager|The controller of the app business logic that implements Agora RTC Engine’s delegates.|Respond to join channel and leave channel; handles callback events like _didJoinedOfUid_ and _didJoinChannel_.
+
+You should see the code follows the logic flow described in the [official documentation](https://docs.agora.io/en/video-calling/get-started/get-started-sdk?platform=ios).  
+
+### Essential API Calls
+* Create an Agora RTC engine
+```swift
+        let config: AgoraRtcEngineConfig = AgoraRtcEngineConfig()
+        config.appId = appId
+        config.audioScenario = .gameStreaming
+        config.channelProfile = .liveBroadcasting      
+        let eng = AgoraRtcEngineKit.sharedEngine(with:config , delegate: self)
+```
+* Join a channel
+```swift
+        self.agoraEngine.joinChannel(byToken: token, channelId: channel, info: info, uid: uid)
+```
+* Leave a channel
+```swift
+		self.agoraEngine.leaveChannel(leaveChannelBlock)
+```
+* Dispose of the engine
+```swift
+		AgoraRtcEngineKit.destroy()
+```
+
+### View Update Sequence
+
+```mermaid
+sequenceDiagram
+AgoraManager ->> RTCEngine: JoinChannel
+RTCEngine --) AgoraManager: didJoinChannel
+AgoraManager ->> AgoraManager : add to users
+AgoraManager -) VisionVideoCallView : users observed
+VisionVideoCallView ->> VisionVideoCallView : construct AgoraVideoCanvasView
+RTCEngine ->> AgoraManager: didJoinedOfUid
+AgoraManager ->> AgoraManager : add to users
+RTCEngine ->> AgoraManager: didOfflineOfUid
+AgoraManager -x AgoraManager : remove from users
+AgoraManager ->> RTCEngine: LeaveChannel
+AgoraManager ->> RTCEngine: destroy
+AgoraManager -x AgoraManager : remove all users
+```
+
+
+## Conclusion
```

Comment:
The conclusion should offer a link out to either the Agora Docs or the next relevant piece of content the reader should visit to continue their journey.